### ISSUE TITLE
feature: expect.toMatchObject

### ISSUE THREAD

**Issue by vinicius73** [lukeed/uvu#177]
Sometimes we just want check if a object is partially equal to another object.
In jest we have [expect.toMatchObject](https://jestjs.io/docs/expect#tomatchobjectobject)

This is very useful when you must check a object with the certain dynamics values.
An example of that is when you call a API to create a resource, commonly, we do not know what is the values of `id` and fields like `created_at` and `updated_at`.

There already is a way to do it with uvu?
If not, there is any plan?

**Comment by hanayashiki** [comment]
Do you mean https://nodejs.org/api/assert.html#assertdeepstrictequalactual-expected-message

**Comment by lukeed** [comment]
No, the `expect.toMatchObject` is a partial match. uvu already includes `assert.equal` which is a deep, equality check for exactness. Here's the difference:

```js
let full = {
  name: 'lukeed',
  human: true,
}

assert.equal({ human: true }, full); // => throws error
assert.match({ human: true }, full); //=> OK, partial match
```

If this were to land, it'd be part of the `assert.match` helper, as shown, since that is uvu's "partial / wilcard matcher" helper.
